lib/translationable.rb

```diff
@@ -0,0 +1,25 @@
+module Translationable
+  def check_translation(user_translation)
+    distance = Levenshtein.distance(full_downcase(self.translated_text),
+                                    full_downcase(user_translation))
+
+    sm_hash = SuperMemo.algorithm(self.interval, self.repeat, self.efactor, self.attempt, distance, 1)
+
+    check_distance(distance, sm_hash)
+  end
+
+  private
+
+  def check_distance(distance, sm_hash)
+    state = true
+    if distance <= 1
+      sm_hash.merge!({ review_date: Time.zone.now + self.interval.to_i.days, attempt: 1 })
```

Comment:
Line is too long. [90/80]<br>Redundant curly braces around a hash parameter.<br>Redundant `self` detected.